go-controller/pkg/ovn/egressip.go

```diff
@@ -2796,32 +2866,43 @@ func createDefaultNoReroutePodPolicies(nbClient libovsdbclient.Client, v4Cluster
 	return nil
 }
 
-// createDefaultNoRerouteNodePolicies ensures egress pods east<->west traffic with hostNetwork pods,
+func (oc *DefaultNetworkController) ensureDefaultNoRerouteNodePolicies() error {
+	oc.eIPC.nodeIPUpdateMutex.Lock()
+	defer oc.eIPC.nodeIPUpdateMutex.Unlock()
+	return ensureDefaultNoRerouteNodePolicies(oc.nbClient, oc.addressSetFactory, oc.controllerName, oc.watchFactory.GetNodes)
+}
+
+// ensureDefaultNoRerouteNodePolicies ensures egress pods east<->west traffic with hostNetwork pods,
 // i.e: ensuring that an egress pod can still communicate with a hostNetwork pod / service backed by hostNetwork pods
 // without using egressIPs.
 // sample: 101 ip4.src == $a12749576804119081385 && ip4.dst == $a11079093880111560446 allow pkt_mark=1008
-func CreateDefaultNoRerouteNodePolicies(nbClient libovsdbclient.Client, addressSetFactory addressset.AddressSetFactory, node *kapi.Node, controllerName string) error {
-	v4NodeAddr, v6NodeAddr := util.GetNodeInternalAddrs(node)
+// Since this function is meant to be used to consider all the IPs of all the nodes, some of which might be dynamic
+// and move among them, it needs to be called under lock.
+func ensureDefaultNoRerouteNodePolicies(nbClient libovsdbclient.Client, addressSetFactory addressset.AddressSetFactory, controllerName string, nodeList func() ([]*v1.Node, error)) error {
+	nodes, err := nodeList()
+	if err != nil {
+		return err
+	}
+
+	v4NodeAddrs, v6NodeAddrs, err := util.GetNodeAddresses(config.IPv4Mode, config.IPv6Mode, nodes...)
+	if err != nil {
+		return err
+	}
+
+	allAddresses := make([]net.IP, 0, len(v4NodeAddrs)+len(v6NodeAddrs))
+	allAddresses = append(allAddresses, v4NodeAddrs...)
+	allAddresses = append(allAddresses, v6NodeAddrs...)
+
 	var as addressset.AddressSet
-	var err error
 	dbIDs := getEgressIPAddrSetDbIDs(NodeIPAddrSetName, controllerName)
 	if as, err = addressSetFactory.GetAddressSet(dbIDs); err != nil {
 		return fmt.Errorf("cannot ensure that addressSet %s exists %v", NodeIPAddrSetName, err)
 	}
-	if v4NodeAddr != nil {
-		// add the nodeIP to the nodeIP address-set
-		if err = as.AddIPs([]net.IP{v4NodeAddr}); err != nil {
-			return fmt.Errorf("unable to add nodeIPs %s/%s for node %s: to the address set %s, err: %v",
-				v4NodeAddr.String(), v6NodeAddr.String(), node.Name, NodeIPAddrSetName, err)
-		}
-	}
-	if v6NodeAddr != nil {
-		// add the nodeIP to the nodeIP address-set
-		if err = as.AddIPs([]net.IP{v6NodeAddr}); err != nil {
-			return fmt.Errorf("unable to add nodeIPs %s/%s for node %s: to the address set %s, err: %v",
-				v4NodeAddr.String(), v6NodeAddr.String(), node.Name, NodeIPAddrSetName, err)
-		}
+
+	if err = as.SetIPs(allAddresses); err != nil {
+		return fmt.Errorf("unable to set IPs to egressip no re-route address set %s: %w", NodeIPAddrSetName, err)
```

Comment:
```suggestion
		return fmt.Errorf("unable to set node IPs: %v to address set %s: %w", allAddresses, NodeIPAddrSetName, err)
```

Reply:
I intentionally left the addresses out because there could be many addresses. Let me know if you still think we should have it anyway.

Reply:
If the libovsdb op cannot fail due to `allAddresses` value then I am ok leaving it out. I would still like for the message to be updated since it is not only meant to be used with egressip.